pkg/vmimage/ssh.go

```diff
@@ -11,11 +11,76 @@ import (
 )
 
 func (builder *Builder) ssh() error {
-	signer, err := ssh.NewSignerFromKey(builder.SSHKey)
+	session, err := builder.newSSHSession()
+	if err != nil {
+		return err
+	}
+	defer session.Close()
+
+	stdout, err := session.StdoutPipe()
 	if err != nil {
 		return err
 	}
 
+	err = session.Start("sudo tail -F -n +1 /var/lib/waagent/custom-script/download/0/stdout")
+	if err != nil {
+		return err
+	}
+
+	io.Copy(os.Stdout, stdout)
+
+	return nil
+}
+
+func (builder *Builder) scp(files []string) error {
+	for _, file := range files {
+		session, err := builder.newSSHSession()
+		if err != nil {
+			return err
+		}
+		defer session.Close()
+
+		stdoutPipe, err := session.StdoutPipe()
+		if err != nil {
+			return err
+		}
+
+		builder.Log.Infof("download %s", file)
+		f, err := os.OpenFile(file, os.O_APPEND|os.O_WRONLY|os.O_CREATE, 0644)
+		if err != nil {
+			return err
+		}
+		defer f.Close()
+
+		err = session.Start(fmt.Sprintf("sudo cat %s", file))
+		if err != nil {
+			return err
+		}
+
+		io.Copy(f, stdoutPipe)
```

Comment:
you don't want to check the error?

Reply:
or is the session going to give you *all* errors, probably not local file errors..

Reply:
I was thinking the same. Not sure if my thinking is right